app/forms.py

```diff
@@ -0,0 +1,143 @@
+# import re
+from django import forms
+# from PIL import Image
+from django.contrib.auth.models import User
+from django.utils.translation import ugettext_lazy as _
+# from django.core.exceptions import ValidationError
+import time
+
+
+class RegistrationForm(forms.Form):
+
+    username = forms.RegexField(
+        regex=r'^\w+$',
+        widget=forms.TextInput(attrs=dict(required=True, max_length=30)),
+        label=_("Username"),
+        error_messages={'invalid': _("This value must contain only letters, \
+         numbers and underscores.")})
+    password1 = forms.CharField(
+        widget=forms.PasswordInput(
+            attrs=dict(
+                required=True,
+                max_length=4,
+                render_value=False,
+                type='password')),
+        label=_("PIN"))
```

Comment:
Why not make this a RegexField too?
```
>>> print re.match('^\d{4}$', 'asdf')
None
>>> print re.match('^\d{4}$', '12345')
None
>>> print re.match('^\d{4}$', '1234')
<_sre.SRE_Match object at 0x1007c88b8>
>>>
```
it would save you the `clean_password1()` function lower down.